test/driver-test/driver-test.cpp

```diff
@@ -0,0 +1,306 @@
+/*
+* driver-test.cpp
+* Nick DeMarco
+* (c) 2020 sigcpp https://sigcpp.github.io. See LICENSE.MD
+*
+* Attribution and copyright notice must be retained.
+* - Attribution may be augmented to include additional authors
+* - Copyright notice cannot be altered
+* Attribution and copyright info may be relocated but they must be conspicuous.
+*
+* Tester for test-stl-lite driver
+*/
+
+#include <iostream>
+#include <vector>
+#include <string>
+
+#include "../verifiers.h"
+#include "../options.h"
+#include "../tester.h"
+#include "../utils.h"
+
+void driver_test();
+void test_get_options();
+void compare(const Options& received, const Options& expected, const std::string& test_case);
+void parse_cmd_line_and_compare(char* args[], const std::string& cmd_line, const Options& expected, const std::string& test_case);
+
+//unit test for the test driver
+void driver_test()
+{
+	test_get_options();
+}
+
+void parse_cmd_line_and_compare(const std::string& cmd_line, const Options& expected,
+	const std::string& test_case)
+{
+	//this is meant to replicate the main() parameter char* argv[]
+	//size 100 was chosen to ensure enough room for many options in a single command-line.
+	char* args[100];
+	std::vector<std::string> v = split(cmd_line, ' ');
+	std::size_t size = v.size();
+	for (std::size_t i = 0; i < size; ++i)
+		args[i] = v[i].data();
+
+	Options received = get_options(args, size);
+	compare(received, expected, test_case);
+}
+
+
+//unit test for get_options
+void test_get_options()
+{
+	Options o;
+	o.command_name = "array_test";
+
+	using std::string;
+	
+	//default arguments
+	const string cmd_line_0("C:/Libraries/stl-lite/array_test.exe");
+	Options cmd_line_0_expected = o;
+	parse_cmd_line_and_compare(cmd_line_0, cmd_line_0_expected, "cmd_line_0");
+
+	//single name-value option supplied
+	const string cmd_line_1("C:/Libraries/stl-lite/array_test.exe -h no");
+	Options cmd_line_1_expected = o;
+	cmd_line_1_expected.header = false;
+	cmd_line_1_expected.header_text = "";
+	parse_cmd_line_and_compare(cmd_line_1, cmd_line_1_expected, "cmd_line_1");
+
+	const string cmd_line_2("C:/Libraries/stl-lite/array_test.exe -h yes");
+	Options cmd_line_2_expected = o;
+	cmd_line_2_expected.header = true;
+	cmd_line_2_expected.header_text = "Running $suite";
+	parse_cmd_line_and_compare(cmd_line_2, cmd_line_2_expected, "cmd_line_2");
+}
+
+//compares received Options object data members to expected Options object data members
+//see fn declaration of is_true for the behavior of the comparison
```

Comment:
Simplify comment; no need to comment about is_true

```suggestion
//perform pair-wise comparison of members in received options and expected options
```